sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1919,6 +1919,21 @@ class Dataset[T] private[sql](
     Except(planWithBarrier, other.planWithBarrier)
   }
 
+  /**
+   * Returns a new Dataset containing rows in this Dataset but not in another Dataset while
+   * preserving the duplicates.
+   * This is equivalent to `EXCEPT ALL` in SQL.
+   *
+   * @note Equality checking is performed directly on the encoded representation of the data
+   * and thus is not affected by a custom `equals` function defined on `T`.
+   *
+   * @group typedrel
+   * @since 2.4.0
```

Comment:
The feature freeze will come soon, so I'm not sure this will appear in v2.4. cc: @gatorsmile @rxin

Reply:
We are still targeting it to 2.4. If we are unable to make it, we can change it.